### DISCUSSION TITLE
Prevent Tree Node from collapsing

### DISCUSSION THREAD

**Discussion by daddycocoaman** [Textualize/textual#4407] [Q&A] [answered]
I have a `Tree` that I would like to display items with arbitrary amounts of child items. However, some of the parent items are also options that can be selected to process `Tree.NodeSelected` events. When these nodes are clicked, the node will collapse or expand. However, I would like to not have these nodes collapsible. 

I'm pretty sure I can handle this by subclassing Tree and overriding  `Tree._toggle_node()` but would it be possible to make this a first-class feature similar to `node.allow_expand`?

Here is the code I'm currently using:

```python
class NoncollapsibleTree(Tree):
    def _toggle_node(self, node: TreeNode) -> None:
        if not node.allow_expand:
            return

        if not node.is_expanded:
            node.expand()
```

**Comment by TomJGooding** [comment, marked as answer]
I think you can currently achieve this without overriding any methods by setting `expand=True` *but* `allow_expand=False`?

```python
from textual.app import App, ComposeResult
from textual.widgets import Tree


class TreeApp(App):
    def compose(self) -> ComposeResult:
        tree = Tree("Root")
        tree.root.expand()

        node_one = tree.root.add("Node 1", expand=True, allow_expand=False)
        node_one.add_leaf("Node 1.1")
        node_one.add_leaf("Node 1.2")

        node_two = tree.root.add("Node 2", expand=True, allow_expand=False)
        node_two.add_leaf("Node 2.1")
        node_two.add_leaf("Node 2.2")

        yield tree


if __name__ == "__main__":
    app = TreeApp()
    app.run()
```

**Reply by daddycocoaman** [reply]
You know what, I dunno how I looked right at that function and didn't put that together. That works even better since it avoids unnecessary expand cursors. Thanks.
